### PR TITLE
fix broken examples causing `pnpm test` to fail

### PR DESCRIPTION
**Pull request by dimitropoulos**
### Description

On main, `pnpm test` fails (sometimes intermittently, depending on cache scenarios, but all the time without cache).  Why?  Well..

Three of the examples had version 9 lockfiles checked in to git, when version 6 is what's required by their `packageManager` fields.  This causes tests to fail quite frequently ([example](https://github.com/vercel/turborepo/actions/runs/11238389716/job/31243099738)) in many PRs.  This happens because when installing in these repos you get `Lockfile not compatible with current pnpm` errors, and it totally blows away the old lockfile and grabs a new one.  Then, since our Next.js version is specified as `^4.1.1` in these package.jsons, it will resolve to `4.2.15` instead of the old `4.1.1`.  Then, it turns out that this minor update causes this diff in the `next-env.d.ts` file:

```diff
/// <reference types="next" />
/// <reference types="next/image-types/global" />

// NOTE: This file should not be edited
- // see https://nextjs.org/docs/basic-features/typescript for more information.
+ // see https://nextjs.org/docs/app/building-your-application/configuring/typescript for more information.
```

The fact of that diff happening causes turbo to not be "FULL TURBO", which the example tests checks for.  That then causes the test suite to fail because there's a turbo cache miss.

### Testing Instructions

`pnpm test` should now pass.

### CODE REVIEW

File: turborepo-tests/helpers/setup_example_test.sh [hunk 1]
```diff
@@ -55,24 +55,28 @@ mkdir -p ./tmp
...
 # Simulating the user's second run
-$turbo_command >./tmp/grep-me-for-hit.txt
+$turbo_command >./tmp/run-2.txt
 
 # Make sure the user hits FULL TURBO on the second go
-if ! grep -q ">>> FULL TURBO" ./tmp/grep-me-for-hit.txt; then
-  echo "No FULL TURBO was found. Dumping logs:"
-  cat ./tmp/grep-me-for-hit.txt >&2
+if ! grep -q ">>> FULL TURBO" ./tmp/run-2.txt; then
+  echo "[ERROR] No 'FULL TURBO' was found.  That shows that the cache was not fully hit on the second run, which was expected behavior."
```
Comment (tknickman) [inline turborepo-tests/helpers/setup_example_test.sh:77]: At a glance this phrasing reads like the second run cache miss is _expected_. Maybe just re-word to something like:


```suggestion
  echo "[ERROR] No 'FULL TURBO' was found.  That indicates that the second run unexpectedly missed the cache."
```

File: turborepo-tests/helpers/setup_example_test.sh [hunk 2]
```diff
@@ -55,24 +55,28 @@ mkdir -p ./tmp
 echo "/tmp/" >> ".gitignore"
 
 # Simulating the user's first run and dumping logs to a file
+$package_manager --version >./tmp/package_manager_version.txt
 $package_manager_command >./tmp/install.txt 2>&1
-$turbo_command >./tmp/grep-me-for-miss.txt
+$turbo_command >./tmp/run-1.txt
```
Comment (tknickman) [inline turborepo-tests/helpers/setup_example_test.sh:60]: Expected rename?

**Review by tknickman — APPROVED** [review]
Some phrasing suggestions, but looks good!

File: turborepo-tests/helpers/setup_example_test.sh (hunk 2 above)
Comment (dimitropoulos) [reply turborepo-tests/helpers/setup_example_test.sh:60]: yes: I was trying to make it clear.  we're running the same command twice so the output for run 1 goes to `run-1.txt` and the output for run 2 goes to `run-2.txt`.  At least for me, this added clarity to understanding what's going on.  The `grep-me` thing was cute, but I found it hard to follow upon trying to understand what's going on here without any other context or asking anyone.

File: turborepo-tests/helpers/setup_example_test.sh (hunk 1 above)
Comment (dimitropoulos) [reply turborepo-tests/helpers/setup_example_test.sh]: good feedback: updated in 5cd348278bb51744c3afeac32898e6690d34e061 to make it clear that "hitting the cache" isn't the same as seeing a wall full of "cache hit" as long as there's even one "cache miss".
